### ISSUE TITLE
Parsing CSV with escape character before enclosure

### ISSUE THREAD

**Issue by pimjansen** [thephpleague/csv#516]
## Bug Report

Wrong output when php escape character is used before enclosure

| Information | Description |
|--------------|---------|
| Version | 9.8.0 and 9.14.0 |
| PHP version | 8.1.13 |
| OS Platform | Docker php-fpm |

### Summary

When using an escape character the document is not parsed as expected.

### Standalone code, or other way to reproduce the problem

```php
$content = '"A field value";"A testing value with slash\";"Another value"';
$reader = Reader::createFromString($content);
$reader->setDelimiter(';');
$reader->setEnclosure('"');
$reader->setHeaderOffset(0);
dd($reader->getRecords());
```

### Expected result

Expect an array with 3 elements

### Actual result

Elements are mixed (1,2)

## Checks before submitting

* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/thephpleague/csv/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/thephpleague/csv/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by nyamsprod** [comment]
@pimjansen thanks for submitting your issue.

What you are experimenting is an underlying issue in PHP CSV parser that was fixed in PHP7.4

As a side-note, there's an error in your script if you use `setHeaderOffset` no records will be return as you will be using the first row as the header.

If instead you use `setEscape` with the empty character then you will get your expected result.
This feature exists in the library since version 9.2 as explained in the documentation. You can find more information via this link https://csv.thephpleague.com/9.0/connections/controls/#the-escape-character.

TL;DR: here's how you should write your script:

```php
<?php

$content = '"A field value";"A testing value with slash\";"Another value"';
$reader = Reader::createFromString($content);
$reader->setEnclosure('"');
$reader->setDelimiter(';');
$reader->setEscape('');
dd([...$reader]);
```

**Comment by pimjansen** [comment]
Will review thanks! About the header it is a copy paste issue for my actual parser since that does way more.

Thanks for the quick reply!
